.github/CONTRIBUTING.md

```diff
@@ -31,141 +31,468 @@ strongly recommended you get approval/traction for it from our forums before sta
+```
+
+### Use var/name format when declaring variables
+While DM allows other ways of declaring variables, this one should be used for consistency.
+
+### Tabs, not spaces
+You must use tabs to indent your code, NOT SPACES.
+
+(You may use spaces to align something, but you should tab to the block level first, then add the remaining spaces)
+
+### No hacky code
+Hacky code, such as adding specific checks (ex: `istype(src, /obj/whatever)`), is highly discouraged and only allowed when there is ***no*** other option. (
+Protip: 'I couldn't immediately think of a proper way so thus there must be no other option' is not gonna cut it here! If you can't think of anything else, say that outright and admit that you need help with it. Maintainers exist for exactly that reason.)
+
+You can avoid hacky code by using object-oriented methodologies, such as overriding a function (called "procs" in DM) or sectioning code into functions and 
+then overriding them as required.
+
+### No duplicated code
+Copying code from one place to another may be suitable for small, short-time projects, but Paradise is a long-term project and highly discourages this.
+
+Instead you can use object orientation, or simply placing repeated code in a function, to obey this specification easily.
+
+### Startup/Runtime tradeoffs with lists and the "hidden" init proc
+First, read the comments in [this BYOND thread](http://www.byond.com/forum/?post=2086980&page=2#comment19776775), starting where the link takes you.
+
+There are two key points here:
+
+1) Defining a list in the variable's definition calls a hidden proc - init. If you have to define a list at startup, do so in New() (or preferably Initialize()) and avoid the overhead of a second call (Init() and then New())
+
+2) It also consumes more memory to the point where the list is actually required, even if the object in question may never use it!
+
+Remember: although this tradeoff makes sense in many cases, it doesn't cover them all. Think carefully about your addition before deciding if you need to use it.
+
+### No magic numbers or strings
+This means stuff like having a "mode" variable for an object set to "1" or "2" with no clear indicator of what that means. Make these #defines with a name that 
+more clearly states what it's for. For instance:
+````DM
+/datum/proc/do_the_thing(thing_to_do)
+  switch(thing_to_do)
+    if(1)
+      (...)
+    if(2)
+      (...)
+````
+There's no indication of what "1" and "2" mean! Instead, you'd do something like this:
+````DM
+#define DO_THE_THING_REALLY_HARD 1
+#define DO_THE_THING_EFFICIENTLY 2
+/datum/proc/do_the_thing(thing_to_do)
+  switch(thing_to_do)
+    if(DO_THE_THING_REALLY_HARD)
+      (...)
+    if(DO_THE_THING_EFFICIENTLY)
+      (...)
+````
+This is clearer and enhances readability of your code! Get used to doing it!
+
+### Control statements
+(if, while, for, etc)
+
+* All control statements must not contain code on the same line as the statement (`if (blah) return`)
```

Comment:
No space after the `if`?
Also, it would probably be nicer as `if(condition) return`